ethereum_history_api/circuits/lib/src/serialize.nr

```diff
@@ -0,0 +1,72 @@
+use crate::account::AccountWithinBlock;
+use crate::account_with_storage::{StorageWithinBlock, Account};
+use crate::misc::{fragment::Fragment, types::{BYTES32_LENGTH, Bytes32, ADDRESS_LENGTH, Address}};
+use dep::std::unsafe::zeroed;
+
+trait Serialize<LEN> {
+    fn serialize(self) -> [Field; LEN];
+}
+
+global U128_SERIALIZED_LEN = 2;
+
+impl Serialize<U128_SERIALIZED_LEN> for U128 {
+    fn serialize(self) -> [Field; U128_SERIALIZED_LEN] {
+        [self.lo, self.hi]
+    }
+}
+
+impl Serialize<BYTES32_LENGTH> for Bytes32 {
+    fn serialize(self) -> [Field; BYTES32_LENGTH] {
+        let mut data: [Field; BYTES32_LENGTH] = zeroed();
+        for i in 0..BYTES32_LENGTH {
```

Comment:
I think you can do:
```rust
self.map(|e| e as Field);
```

Reply:
Unfortunately - it does not work:
https://github.com/noir-lang/noir/issues/4994